### PR TITLE
REV-260 fix schedule bugs

### PR DESCRIPTION
**Pull request by IbrahimAljabrST**
### Description

fix schedule bugs

### CODE REVIEW

File: src/components/edit-listing-schedule/ListingSchedule/index.jsx [hunk 1]
```diff
@@ -31,6 +34,7 @@ function EditListingSchedule() {
   const [endDate, setEndDate] = useState(null);
   const [updateDate, setUpdateDate] = useState(null);
   const [dateRange, setDateRange] = useState(weekDays);
+  const [scheduleSaved, setScheduleSaved] = useState(false);
```
Comment (mohdabed123) [inline src/components/edit-listing-schedule/ListingSchedule/index.jsx:37]:
```suggestion
  const [savedSchedule, setSavedSchedule] = useState(false);
```

File: src/components/listing-page/ScheduleVisit/schedule-visit.styles.js [hunk 2]
```diff
@@ -67,6 +66,8 @@ export const DateCard = styled.div`
   text-align: center;
   margin: 1rem;
 
+  border: ${({ activeDay }) => activeDay && `0.06rem solid ${colors.green}`};
```
Comment (mohdabed123) [inline src/components/listing-page/ScheduleVisit/schedule-visit.styles.js:69]:
```suggestion
  border: ${({ IsActiveDay }) => activeDay && `0.06rem solid ${colors.green}`};
```

File: src/components/listing-page/ScheduleVisit/index.jsx [hunk 3]
```diff
@@ -27,14 +29,17 @@ import {
  */
 function ScheduleVisit({ data, id }) {
   const [dateRange, setDateRange] = useState([]);
+  const [activeDay, setActiveDay] = useState(null);
   const [selectedDay, setSelectedDay] = useState([]);
   const [filteredDays, setFilteredDays] = useState([]);
   const [selectedHour, setSelectedHour] = useState(null);
   const [requestedDate, setRequestedDate] = useState(null);
+  const [scheduleSaved, setScheduleSaved] = useState(false);
```
Comment (mohdabed123) [inline src/components/listing-page/ScheduleVisit/index.jsx:37]:
```suggestion
  const [savedSchedule, setSavedSchedule] = useState(false);
```

File: src/components/edit-listing-schedule/TimeInput/index.jsx [hunk 4]
```diff
@@ -65,6 +65,8 @@ function TimeInput({
       <DateInputs>
         <Controller
           control={control}
+          // eslint-disable-next-line no-unneeded-ternary
+          defaultValue={time ? time : new Date()}
```
Comment (mohdabed123) [inline src/components/edit-listing-schedule/TimeInput/index.jsx:69]:
```suggestion
          defaultValue={time ?? new Date()}
```

File: src/components/edit-listing-schedule/ListingSchedule/index.jsx [hunk 5]
```diff
@@ -120,35 +125,39 @@ function EditListingSchedule() {
...
-        )
-      );
+    const pickedDaysDate = getDifferenceBetweenTwoDates(startDate, endDate);
+
+    if (pickedDaysDate > 0) {
+      if (
+        startDate &&
+        endDate &&
+        getDifferenceBetweenTwoDates(startDate, endDate, 7)
+      ) {
```
Comment (mohdabed123) [inline src/components/edit-listing-schedule/ListingSchedule/index.jsx:135]: one if statement 
```suggestion

      if (
      pickedDaysDate > 0 &&
        startDate &&
        endDate &&
        getDifferenceBetweenTwoDates(startDate, endDate, 7)
      ) {
```

**Review by mohdabed123 — CHANGES_REQUESTED** [review]

File: src/components/edit-listing-schedule/ListingSchedule/index.jsx (hunk 5 above)
Comment (IbrahimAljabrST) [reply src/components/edit-listing-schedule/ListingSchedule/index.jsx:135]: we need 2 if in this case we have a different else for each one

**Review by mohdabed123 — APPROVED** [review]

**Pull request merged**